docs: add documentation for using lambda functions (#7092)
confluentinc/ksql: docs/concepts/lambda-functions.md

---
layout: page

> title: ksqlDB - Lambda Functions

Reviewer: ```suggestion
title: Lambda Functions
```

> ---

Reviewer: keywords would be great!

# Lambda Functions

> Lambda functions enable users to express simple inline functions that can be applied to input values in various ways. 

Reviewer: ```suggestion
Use lambda functions, or "lambdas" for short, to express simple inline functions that can be applied to input values in various ways.
```

> For example, lambda functions could be applied to each element of a collection, resulting in a transformed output collection. 

Reviewer: ```suggestion
For example, you could apply a lambda function to each element of a collection, resulting in a transformed output collection. 
```

> Lambda functions could also be used to filter the elements of a collection, or reduce a collection down to a single value.

Reviewer: ```suggestion
Also, you can use lambdas to filter the elements of a collection, or reduce a collection to a single value.
```

> This allows users to express user-defined functionality in a way that doesn’t require them to implement full-fledged UDFs.

Reviewer: ```suggestion
The advantage of a lambda is that you can express user-defined functionality in a way that doesn’t require implementing a full [UDF](/how-to-guides/create-a-user-defined-function).
```
